sdk/python/kfp/components/structures_test.py

```diff
@@ -495,8 +497,19 @@ def test_from_container_dict_no_placeholders(self):
         loaded_container_spec = structures.ContainerSpec.from_container_dict(
             container_dict)
 
-
-class TestComponentSpec(unittest.TestCase):
+    def test_validate_command_and_arg(self):
```

Comment:
Because this is a slightly more complex validator than other validators, WDYT about testing this using a real pipeline definition that references an artifact directly (either in addition or in place of this test -- up to you)? In addition to providing a more realistic test, it will help document the undesirable pipeline authoring as well.

Reply:
I put a test that tries to replicate the real authoring experience at `container_component_artifact_channel_test.py`, and I think I should move that test into structures_test.py now for more readability. Thanks for the suggestion!
P.S. Line 498 and 499 are deleted by mistake: I will correct that in the next commit.

Reply:
Ah, that's right. Thanks for reminding me about those tests.